### DISCUSSION TITLE
Cant Enable Only one OPEN Stream on a file

### DISCUSSION THREAD

**Discussion by MagaKhachatryan** [dokan-dev/dokan-dotnet#319] [Q&A] [answered]
Hi. I have a problem here. I don't want many streams to be opened for the same file at the same time. So I use semaphores. The semaphore waits before opening the stream in CreateNew function in Mirror and is Released in Close and Cleanup functions where streams are Closed. 
Copying  the file into our Drive works fine but when I try to open it ,it can open only .txt file but it cant open for example the Word file or ppt file .
When I try to log the process I see that after some actions a stream is opened and is not closed and new streams that must be opened are waiting and after some time it says that Word experienced error trying to open it and I think it is the result of timeout.
But when I delete the semaphore it works fine and when I look at logs I see that a few threads are opened for the same file at the same time.

So my question is. Can a word file or ppt or another microsoft file that is in our Drive be opened having only one open stream at the same time?

**Comment by Liryna** [comment, marked as answer]
Hi @MagaKhachatryan ,

This is a very hard requirement to achieve to only have one stream open on one file at a time.
It is not only word that will try to access the word file but any application on the system which no one has control one.

Depending on the reason why having the lock in Create, I would suggest reducing the locking scope to only where you need it (like write and read if it is to protect data).

**Reply by MagaKhachatryan** [reply]
I got it @Liryna , thank you!
